server/datastore/mysql/hosts.go

```diff
@@ -1615,7 +1615,11 @@ WHERE
 	}
 
 	if err := sqlx.GetContext(ctx, ds.reader, &row, query, args...); err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			return &fleet.OSVersions{}, nil
```

Comment:
I think we should follow the other pattern we've been following:

```
return nil, ctxerr.Wrap(ctx, notFound("OSVersions"))
```

And handle it appropriately on the caller for this.